src/tree/v2/BranchDataItem.ts

```diff
@@ -18,6 +18,11 @@ export class BranchDataItem implements ResourceGroupsItem, Wrapper {
         itemCache.addBranchItem(this.branchItem, this);
     }
 
+    /** Needed for tree item picker PickAppResourceStep */
+    public get resource(): ApplicationResource | undefined {
```

Comment:
If we don't already have it, I think there ought to be an interface for the RG items specifically at the AppResource level, where they will have this `resource` property. i.e.:
```typescript
interface AppResourceItem extends ResourceGroupsItem {
    readonly resource: ApplicationResource | undefined; // TODO: can `undefined` even happen?
}
```

Reply:
I've added `ApplicationResourceBranchDataItem` which handles special app resource behavior.